### PR TITLE
Initial Italian Translation

### PR DESCRIPTION
**Pull request by llegovich**
This is the initial Italian translation for Habot.
Signed-off-by: Luciano Legovich <llegovich@yahoo.it>

### CODE REVIEW

**Comment by ghys** [comment]
@llegovich the travis test is failing because the training data is invalid (click on Details below and expand the `$ tail -n 2000 .build.log` line then scroll down):
```
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   TrainerItTest.testIT:22 » IO Found unexpected annotation: dell'<START:object> ...
```
This means all tags like `<START:object>` must be preceded by a space, you cannot have an apostrophe immediately before it. You need to find and replace all occurrences. In fact you can get rid of the apostrophes because they will be removed anyways by the tokenizer. Look at the French training data for similar examples.

**Comment by llegovich** [comment]
Hi @ghys, just removed apostrophes and corrected some typos. This time I didn't use git web ui but VS Code, hope it is signed off correctly.

**Comment by llegovich** [comment]
Hi, not able to get rid of this "unexpected annotation" error.
Any help?
thx

File: bundles/org.openhab.ui.habot/src/main/resources/train/it/set-value.txt [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+metti il <START:object> volume della musica <END> in <START:location> cucina <END> al <START:value> 70 % <END>
+cambia il colore delle <START:object> luci <END> del <START:location> bagno <END> in <START:color> giallo <END>
+cambia il colore delle <START:object> luci <END> del <START:location> primo piano <END> in <START:color> bianco <END>
+maetti le <START:object> luci <END> della <START:color> camera <END> al <START:value> 30% <END>
+imposta il <START:object> riscaldamento <END> dello <START:location> studio <END> a <START:color> 70°F <END>
+imposta le <START:object> luci <END> a <START:color> rosso <END>
+imposta le <START:object> luci <END> to <START:color> arancione <END>
+imposta le <START:object> luci <END> nel <START:location> soggiorno <END> a <START:color> blu <END>
+imposta le <START:object> luci <END> della <START:location> piscina <END> a <START:color> rosso <END>
+imposta le <START:object> luci<END> del <START:location> giardino <END> a <START:color> viola <END>
```
Comment (ghys) [inline bundles/org.openhab.ui.habot/src/main/resources/train/it/set-value.txt]: Should be because of the missing space after "luci" here.

**Comment by ghys** [comment]
@llegovich assertion failed in tests are usually due to a lack of training data. The tuning phase took me some time in English and French ;) Usually what I did is add the failing sentence from the test, properly tagged, to the relevant training file and see what happens (sometimes it makes another assertion fail elsewhere...) - it's always better to run tests in an IDE instead of relying on Travis log (which are truncated). Or if you can't make some assertions work, give up and comment the line in the test code...

**Comment by llegovich** [comment]
Hi @ghys , could you please have a look at the last build fail (build 24)? I'm not sure I understand the errors. thx

**Comment by ghys** [comment]
To be honest, neither do I! 😆

**Comment by llegovich** [comment]
@kaikreuzer , @wborn can you help with the builds? I don't understand the errors... thx

**Comment by ghys** [comment]
@llegovich my apologies for the lack of follow-ups from my side lately...
I think whatever made the build suddenly fail were fixed in #64.
So if you kindly could [rebase](https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request) this PR to include the latest changes to master, that would be great and I believe it would fix the issues with the build, so we can at last merge your work!
Thanks!

**Comment by wborn** [comment]
The Travis build should work again @llegovich now that https://github.com/openhab/openhab-webui/pull/70 got merged. So hopefully this PR also gets build successfully when you re-trigger the build by rebasing on the current master or after pushing a new commit.

**Comment by llegovich** [comment]
Yes, it looks better now.
The lesson I learnt is that probably the training material is not enough, so I'm keen to see this in action on the next build and test it to be able to enrich the training scripts.
Thanks a lot for your support! :-)

**Comment by ghys** [comment]
That's great to hear! :)
Indeed getting the training right is not trivial, the tests are supposed to guarantee a "decent enough" performance. The idea to improve it little by little is basically this: if you encounter a misinterpretation during real-world usage and it definitely should have worked, you add the sentence to the unit test and alter the training data until it passes (and there is no regression elsewhere).
I know it's not so easy to run them locally right now, waiting for Travis to finish to check if they succeeded is not ideal.

**Comment by ghys** [comment]
@llegovich I almost forgot 😓:  you will want to also submit a PR to openhab-core, adding translations for the semantic tags (a new _tags_it.properties_ file here: https://github.com/openhab/openhab-core/tree/master/bundles/org.openhab.core.semantics/src/main/resources). This will ensure you can refer to semantically-tagged items with Italian terms.

Fortunately it's almost similar to https://github.com/openhab/openhab-webui/blob/master/bundles/org.openhab.ui.habot/src/main/resources/tagattributes_it.properties - which should be removed from here someday because it will not be used by default anymore.

**Comment by ghys** [comment]
@llegovich Actually running the tests locally is simple enough with Visual Studio Code (see the complete setup instructions at https://www.openhab.org/docs/developer/ide/vscode.html), here's a short tutorial:
1. Download & install VS Code from https://code.visualstudio.com/ and make sure you have a JDK installed (https://www.oracle.com/technetwork/java/javase/downloads/jdk8-downloads-2133151.html) and Git.
2. Open the _bundles/org.openhab.ui.habot_ of your local fork in VS Code, make sure you have the latest version - click on 
image
on the let then pull the latest changes:
image
3. On the bottom-left corner, click on master then create a new branch:
image
and give it a name
4. Go to Extensions on the left
image
search for "java pack" and install the _"Java Extensions Pack"_ extension. Follow the eventual messages and instructions then restart VS Code.
5. You should normally get a new Test icon on the left. The tests should also appear if the right folder is opened. Find the TrainerItTest and click on it to open the code
image
You can also open the Output window with the _View > Output_ menu option. In the output window, there is a combobox, choose "Java Test Runner":
image
6. In the code, click on Run Test above `public class ...`. The test should run and pass (there will be a ✔️).
If the test fails, you will see a ❌ and you can click on it to open the results:
image
7. Once you're done you can commit, sign-off and push the changes to your fork with VS Code too.

Hope it helps!

**Pull request merged**